libnopegl/src/scene.c

```diff
@@ -267,6 +267,27 @@ int ngl_scene_get_filepaths(struct ngl_scene *s, char ***filepathsp, size_t *nb_
     return 0;
 }
 
+static void update_file_back_ref(struct ngl_scene *s, size_t index, const char *str)
+{
+    const char **filep = ngli_darray_get(&s->files, index);
+    *filep = str;
+}
+
+void ngli_scene_update_file_back_ref(struct ngl_node *node, const struct node_param *par)
+{
+    struct ngl_scene *s = node->scene;
+    const uint8_t **pars = ngli_darray_data(&s->files_par);
+    for (size_t i = 0; i < ngli_darray_count(&s->files_par); i++) {
+        const uint8_t *base_ptr = node->opts;
+        const uint8_t *parp = base_ptr + par->offset;
+        if (pars[i] == parp) {
+            char *str = *(char **)parp;
+            update_file_back_ref(s, i, str);
+            break;
+        }
+    }
```

Comment:
Maybe return early and assert if par is not found or return an error ?

Reply:
Added an assert